### ISSUE TITLE
[NBug] Exception of type 'System.OutOfMemoryException' was thrown.

### ISSUE THREAD

**Issue by e2m32** [gitextensions/gitextensions#8492]
:warning: The sections below must be filled in and this text must be removed or the issue will be closed.

## Current behaviour
 Program crashed due to memory error.

## Expected behaviour
 Program displays differences in file before committing. 

## Steps to reproduce

 Add large file to repository, commit. Add file to be tracked by LFS. View changes and attempt to see differences in file.

## Error Details
```
System.OutOfMemoryException: Exception of type 'System.OutOfMemoryException' was thrown.
   at System.String.SplitInternal(Char[] separator, Int32 count, StringSplitOptions options)
   at GitUI.Editor.Diff.DiffLineNumAnalyzer.Analyze(String diffContent)
   at GitUI.Editor.Diff.DiffViewerLineNumberControl.DisplayLineNumFor(String diff)
   at GitUI.Editor.FileViewerInternal.SetText(String text, Action openWithDifftool, Boolean isDiff)
   at GitUI.Editor.FileViewer.<>c__DisplayClass103_0.<ViewPatchAsync>b__0()
   at GitUI.Editor.FileViewer.<llShowPreview_LinkClicked>b__143_0()
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.ExecuteJob[T](Func`1 asyncMethod, JoinableTask job)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()
   at GitUI.Editor.FileViewer.llShowPreview_LinkClicked(Object sender, LinkLabelLinkClickedEventArgs e)
   at System.Windows.Forms.LinkLabel.OnLinkClicked(LinkLabelLinkClickedEventArgs e)
   at System.Windows.Forms.LinkLabel.OnMouseUp(MouseEventArgs e)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.Label.WndProc(Message& m)
   at System.Windows.Forms.LinkLabel.WndProc(Message& msg)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

## Additional information
File is large >126mb. Just added it to LFS, clicked the "This file is <...> Click to show anyway" link to view differences, this caused the crash.

## Environment
- Git Extensions 3.4.3.9999
- Build d4b0f48bbf3e39a71f5d7ff5231be5678d4aa0f1
- Git 2.23.0.windows.1 (recommended: 2.25.1 or later)
- Microsoft Windows NT 10.0.17134.0
- .NET Framework 4.8.4240.0
- DPI 130dpi (135% scaling)

**Comment by gerhardol** [comment]
Related to #7146

A try-catch should wrap the diff view, not much GE can do

**Comment by wischi-chr** [comment]
I'm pretty sure you can't just try-catch an out-of-memory exception

**Comment by gerhardol** [comment]
> Ok for some reason I thought that OutOfMemory is a corrupted state exception, which it's not, so it might be recoverable.

A better solution anyway would be to not analyze contents that is too big, to avoid the exception.

**Comment by gerhardol** [comment]
If this is caused in the same as #8634, it was fixed in #8661
Previous occurrences with explanation has been related to use of UTF-7 (that is removed in 3.5)

Possibly a duplicate of #7146

**Comment by RussKie** [comment]
Presume fixed. If not, please provide a standalone repro on the latest versions, and we'll reopen.
